src/librustc/ty/structural_impls.rs

```diff
@@ -95,8 +96,11 @@ impl fmt::Debug for ty::BoundRegion {
         match *self {
             ty::BrAnon(n) => write!(f, "BrAnon({:?})", n),
             ty::BrNamed(did, name) => {
-                write!(f, "BrNamed({:?}:{:?}, {})",
-                        did.krate, did.index, name)
+                if did.index == CRATE_DEF_INDEX {
+                    write!(f, "BrNamed({})", name)
+                } else {
+                    write!(f, "BrNamed({:?}, {})", did, name)
```

Comment:
Not sure if this is easily done, but can you eliminate the `Defid(foobar)` wrapping? Just the content seems fine since this is already wrapped in a BrNamed

Reply:
It's rare enough that I don't want to touch that right now.
Ideally this would print a proper path, and it still probably could using `ty::tls::with` or w/e, but also `DefId` printout could be better in general.